tensorflow/python/ops/rnn.py

```diff
@@ -611,22 +622,7 @@ def _dynamic_rnn_loop(cell,
   inputs_got_shape = tuple(input_.get_shape().with_rank_at_least(3)
                            for input_ in flat_input)
 
-  const_time_steps, const_batch_size = inputs_got_shape[0].as_list()[:2]
-
-  for shape in inputs_got_shape:
-    if not shape[2:].is_fully_defined():
```

Comment:
Since we want this to run even when the static rank is unknown, these checks should be removed, right? As also discussed [here](https://github.com/tensorflow/tensorflow/pull/5142#issuecomment-260748149).